### PR TITLE
DROOLS-1023 Change kmodule.xsd targetNamespace to become versionless …

### PR DESCRIPTION
**Pull request by ge0ffrey**
DROOLS-1023 Change kmodule.xsd targetNamespace to become versionless (to avoid hardcoding "6.0.0" in it)

### CODE REVIEW

**Comment by ge0ffrey** [comment]
The PR build fails because the droolsjbpm-knowledge PR's changed weren't applied.
I didn't run all drools tests locally though. But I verified it worked with the new and the old namespace (although there's no test currently for the old namespace...)

**Comment by psiroky** [comment]
The changes from droolsjbpm-knowledge were applied, looking at the build log these upstream repositories were cloned and build before this one:

```
git://github.com/droolsjbpm/droolsjbpm-build-bootstrap:master
git://github.com/ge0ffrey/droolsjbpm-knowledge:kmodule-xsd-patch
```

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieBuilderTest.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public void testResourceInclusion() {
                 "end\n";
 
         String kmodule = "<kmodule xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" \n" +
-                "         xmlns=\"http://jboss.org/kie/6.0.0/kmodule\">\n" +
+                "         xmlns=\"http://www.drools.org/xsd/kmodule\">\n" +
```
Comment (mariofusco) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieBuilderTest.java:67]: @ge0ffrey I think what you did broke backward compatibility anyway. The old namespace should still work but if I try to use the old one here this test fails.

Comment (ge0ffrey) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieBuilderTest.java:67]: That test should indeed work with the old namespace. Let's duplicate that
test so the old namespace is tested too (as well as the new one).

I am on PTO today, I'll try to take a look tomorrow

2016-01-12 20:06 GMT+01:00 Mario Fusco notifications@github.com:

> In
> drools-compiler/src/test/java/org/drools/compiler/integrationtests/KieBuilderTest.java
> https://github.com/droolsjbpm/drools/pull/596#discussion_r49498241:
> 
> > @@ -64,7 +64,7 @@ public void testResourceInclusion() {
> >                  "end\n";
> > 
> > ```
> >      String kmodule = "<kmodule xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" \n" +
> > ```
> > -                "         xmlns=\"http://jboss.org/kie/6.0.0/kmodule\">\n" +
> > -                "         xmlns=\"http://www.drools.org/xsd/kmodule\">\n" +
> 
> @ge0ffrey https://github.com/ge0ffrey I think what you did broke
> backward compatibility anyway. The old namespace should still work but if I
> try to use the old one here this test fails.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/droolsjbpm/drools/pull/596/files#r49498241.

## 

With kind regards / Met vriendelijke groeten,
Geoffrey De Smet

**Comment by ge0ffrey** [comment]
@mariofusco there were some serious issues with my PR, but they should be all solved now.
I've added 3 unit tests to test the new schema, the old schema and an invalid schema.
The original valid code in KieModuleModelImpl didn't actually work (despite that I swear it worked for me when using it in OptaPlanner with the old and new schema). In any case, I fixed it to work, but unfortunately this required duplicating the stream initialization (which isn't a real performance cost anyway). See changes on KieModuleModelImpl in my last commit for more info.

**Comment by mariofusco** [comment]
Merged by https://github.com/droolsjbpm/drools/commit/2149530174e121a437bbef93196a4eef9121f9fd

**Pull request closed without merging**